spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/provisioning/KafkaTopicProvisioner.java

```diff
@@ -231,7 +230,10 @@ public Object doWithRetry(RetryContext context) throws RuntimeException {
 		}
 	}
 
-	public Collection<PartitionInfo> getPartitionsForTopic(final int partitionCount, final Callable<Collection<PartitionInfo>> callable) {
+	public Collection<PartitionInfo> getPartitionsForTopic(final int partitionCount, 
+			UnexpectedPartitionCountHandling partitionCountExpectationHandling,
```

Comment:
Compile errors. This has to be declared as final as an inner class is accessing this variable.